backend/forms/graphql/response.py

```diff
@@ -61,18 +63,69 @@ def resolve_created_by(response: Response, info) -> User | None:
         description=resolve_created_by.__doc__,
     )
 
+    @staticmethod
+    def resolve_cached_dimensions(response: Response, info, key_dimensions_only: bool = False):
+        """
+        Returns the dimensions of the response as
+        a dict of dimension slug -> list of dimension value slugs. If the response
+        is not related to a survey, there will be no dimensions and an empty dict
+        will always be returned.
+
+        Using this field is more efficient than querying the dimensions field
+        on the response, as the dimensions are cached on the response object.
+        """
+        cached_dimensions = response.cached_dimensions
+
+        if key_dimensions_only:
+            survey = response.survey
+            if survey is None:
+                return {}
```

Comment:
Soft opinion: I would probably raise instead of returning nothing here, since it sounds like a caller problem to request key dimensions for a key-dimension-less response.

Reply:
similarly it makes questionable sense to ask for any dimensions for a surveyless form, but that will gracefully degrade into `{}` and throwing would require a separate check, so I figured with `keyDimensionsOnly: true` it should work similarly

Reply:
hey I found a clever way to elide the explicit check on survey

Reply:
https://github.com/con2/kompassi/pull/370/files#diff-603d354879460e03dae815e51cf4a2bea9dc0d424d34f49be3a3d6e03b291796R79-R85